frontend/src/views/AppView.vue

```diff
@@ -210,12 +210,20 @@ const copyURL = () => {
 };
 
 const config = ref(searchConfigStore.searchConfig);
+const loadingBar = useLoadingBar();
 watch(
   config, // ref to searchConfigStore.searchConfig
   () => {
+    state.dataRow = [];
     state.isLoading = true;
-    refreshDataTable();
-    state.isLoading = false;
+    loadingBar.start();
+    setTimeout(() => {
```

Comment:
setTimeout(() => {
      refreshDataTable();
      state.isLoading = false;
    }, 0);

Reply:
I guess this might work since what we need to do is to send it async so we give the framework a chance to re-render the view based on state.isLoading = true;

Reply:
oh, actually a better way is to use nextTick